overlord/snapstate/snapmgr.go

```diff
@@ -202,6 +202,9 @@ type ComponentSetup struct {
 	// DownloadInfo contains information about how to download this component.
 	// Will be nil if the component should be sourced from a local file.
 	DownloadInfo *snap.DownloadInfo `json:"download-info,omitempty"`
+	// Flags is a set of flags that control the behavior of the component
+	// installation/update.
+	Flags componentInstallFlags `json:"flags"`
```

Comment:
componentInstallFlags itself doesn't seem to have json tags, also should flags here be embedded like they are in SnapSetup?